hca/dss/util/__init__.py

```diff
@@ -51,3 +55,13 @@ def hardlink(source, link_name):
         # Check device first because different drives can have the same inode number
         if source_stat.st_dev != dest_stat.st_dev or source_stat.st_ino != dest_stat.st_ino:
             raise
+
+
+@contextlib.contextmanager
```

Comment:
This seems to conflate `FileExistsError`s raised by the body of surrounding `with` statement (which shouldn't be suppressed) with those raised by the CM itself. Looking at the code of atomicwrites it is not clear how we could fix this since it uses `contextlib.contextmanager` not the more direct approach of implementing `__enter__` and `__exit__` methods.

Reply:
Wow, nice catch. I think I have a solution...